test/named-parameter-order-test.js

```diff
@@ -0,0 +1,36 @@
+var shuffle = require('shuffle-array');
+var test = require('tap').test;
+var docuri = require('..');
+
+var simulateWrongOrderObjectKeys = function(cb){
+  var orig = Object.keys;
+  Object.keys = function(obj){
+    var keys = orig(obj);
+    var real = keys.toString();
+    while (real === keys.toString())
+      shuffle(keys);
```

Comment:
For now, yes. But I can imagine how someone will "fix" this by using `Object.keys` once to remember the "correct" order and use that from a variable later assuming that a later `Object.keys` invocation returns the same value, when it doesn't.